lib/webpacker/manifest.rb

```diff
@@ -13,32 +13,95 @@ def file_path
       Webpacker::Configuration.manifest_path
     end
 
+    # Throws an error if the file is not found. If Configuration.compile? then compilation is invoked
+    # the file is missing.
+    # React on Rails users will need to set Configuration.compile? to false as compilation is configured
+    # in the package.json for React on Rails.
     def lookup(name)
       if Webpacker::Configuration.compile?
         compile_and_find!(name)
       else
+        # Since load_instance is a no-op for production, this should be fine.
+        # If we're using `webpack -w` to recompile when files changes, we need
+        # to call this before find!
+        load_instance
         find!(name)
       end
     end
 
+    # Why does this method exist? Testing? It's not in the README
     def lookup_path(name)
-      Rails.root.join(File.join(Webpacker::Configuration.public_path, lookup(name)))
+      Rails.root.join(File.join(Webpacker::Configuration.output_path, lookup(name)))
     end
 
-    private
-      def find!(name)
-        raise Webpacker::FileLoader::FileLoaderError.new("Webpacker::Manifest.load must be called first") unless instance
-        instance.data[name.to_s] || raise(Webpacker::FileLoader::NotFoundError.new("Can't find #{name} in #{file_path}. Is webpack still compiling?"))
+    # Helper method to determine if the manifest file exists. Maybe Webpack needs to run?
+    # **Used by React on Rails.**
+    def exist?
+      path_object = Webpacker::Configuration.manifest_path
+      path_object.exist?
+    end
+
+    # Find the real file name from the manifest key. Don't throw an error if the file is simply
+    # missing from the manifest. Return nil in that case.
+    # If no manifest file exists, then throw an error.
+    # **Used by React on Rails.**
```

Comment:
In order to put this logic into React on Rails, we'd have to expose even more to the outside world.

I'd suggest we have the methods `find!` and `lookup_path` take an optional named param (or hash param) of `throw: true` and React on Rails can call them passing `throw: false`.